### ISSUE TITLE
AlbumList view query is very slow

### ISSUE THREAD

**Issue by gryphonmyers** [BinaryBrain/Beets-SubSonic-API#12]
The AlbumList view is quite slow to render, stemming from the fact that the handler pulls ALL albums out of the Beets DB then does pagination in memory. For large libraries, it is very noticeable how slow this is - trying to navigate a paginated view takes forever.

Solution - we can't do pagination using built-in Beets queries, but we can should be able to use lib.transaction to perform our own SQLite query. Instead of paginating in memory we would paginate with built-in sqlite limits and offsets.

**Comment by BinaryBrain** [comment]
Yes, I totally agree. I took some shortcut to have a working version but this is the kind of optimization I need to do next.

**Comment by gryphonmyers** [comment]
If I can find some time I am also willing to contribute

**Comment by FlorentLM** [comment]
Hey hey,
Actually implemented something like that in [my fork](https://github.com/FlorentLM/Beetstream) (along with some other changes) - I might create a pull request if you are intersted :)

**Comment by BinaryBrain** [comment]
@FlorentLM Your fork looks great! You can definitely create a PR and I'll merge it after testing.
I don't know if you also tested with these [supported clients](https://github.com/BinaryBrain/Beetstream?tab=readme-ov-file#supported-clients), but I encourage you to do so to avoid breaking anyone's experience.

**Comment by FlorentLM** [comment]
I tried with several clients but ...none in this list haha
But yes I will definitely do more extensive testing before any PR :)
